### PR TITLE
fix warnings and closing unused connection

### PR DESCRIPTION
**Pull request by caijun**
Warning messages:
```
1: In file(con, "r") :
  cannot open file '/home/travis/build/tdhock/animint/tests/testthat/pids.txt': No such file or directory
2: closing unused connection 5 (/home/travis/build/tdhock/animint/tests/testthat/pids.txt) 
3: closing unused connection 4 (/home/travis/build/tdhock/animint/tests/testthat/pids.txt) 
4: closing unused connection 3 (/home/travis/build/tdhock/animint/tests/testthat/pids.txt) 
````
and 

```
Warning: bad markup (extra space?) for make_text.Rd
Watning: unexpected section header '\value' in make_text.Rd
```

There is still error message `Undefined error in RCurl call` from RSelenium, but it doesn't have any influence to our functions of testting.

### CODE REVIEW

File: R/geoms.R [hunk 1]
```diff
@@ -122,7 +122,7 @@ make_bar <- function(data, x.name, alpha=1){
 #' @param x x coordinate of label position
 #' @param y y coordinate of label position
 #' @param label.var variable matching showSelected, used to obtain label value
-#' @param format String format for label. Use %d, %f, etc. to insert relevant label.var value.
+#' @param format String format for label. Use \%d, \%f, etc. to insert relevant label.var value.
 #' @return a geom_text layer.
```
Comment (cpsievert) [inline R/geoms.R]: Did you run `devtools::document()` (or similar)? I don't see any updates to Rd files.

File: R/testHelpers.R [hunk 2]
```diff
@@ -122,8 +126,8 @@ run_servr <- function(directory = ".", port = 4848,
                       code = "servr::httd(dir='%s', port=%d)") {
   dir <- normalizePath(directory, winslash = "/", mustWork = TRUE)
   cmd <- sprintf(
-    paste("library(methods); write.table(Sys.getpid(), file='%s', append=T, row.name=F, col.names=F);", code),
-    pid_file(), dir, port
+    paste("write.table(Sys.getpid(), file='%s', append=T, row.name=F, col.names=F);", code),
```
Comment (cpsievert) [inline R/testHelpers.R]: Is there a good reason for not loading __methods__? I think it's safer to leave it in there.

File: R/testHelpers.R [hunk 3]
```diff
@@ -101,12 +101,16 @@ tests_run <- function(dir = ".", filter = NULL) {
...
+  f <- file.path(find_test_path(), "pids.txt")
+  if (file.exists(f)) {
+    e <- try(readLines(con <- file(f), warn = FALSE), silent = TRUE)
+    if (!inherits(e, "try-error")) {
+      pids <- as.integer(e)
+      res <- c(res, tools::pskill(pids))
+    }
+    close(con)
+    unlink(f)
   }
```
Comment (cpsievert) [inline R/testHelpers.R]: nice!!

File: R/geoms.R (hunk 1 above)
Comment (caijun) [inline R/geoms.R]: Without `\`, build animint will produce following warnings

```
Warning: /Users/tonytsai/Documents/R/packages/animint/man/make_text.Rd:20: unexpected section header '\value'
Warning: /Users/tonytsai/Documents/R/packages/animint/man/make_text.Rd:23: unexpected section header '\description'
Warning: /Users/tonytsai/Documents/R/packages/animint/man/make_text.Rd:26: unexpected section header '\author'
Warning: /Users/tonytsai/Documents/R/packages/animint/man/make_text.Rd:30: unexpected END_OF_INPUT '
```

File: R/testHelpers.R (hunk 2 above)
Comment (caijun) [inline R/testHelpers.R]: I found without loading library `methods` it also works. You don't even have to do that, as `methods` is already loaded by default in the R session.

**Comment by caijun** [comment]
I keep failing for building on Travis-CI due to `Error: animint2gist() returns an object of class 'gist'`.

The check information tells me to 

```
In non-interactive environments, please set GITHUB_PAT env to a GitHub access token (https://help.github.com/articles/creating-an-access-token-for-command-line-use)
```

but I don't know where to set GITHUB_PAT.

**Comment by cpsievert** [comment]
Ah, sorry, I think for security reasons it can't access GITHUB_PAT because this is sent from your fork of animint. Do you have write access to animint?

**Comment by caijun** [comment]
I don't know how to check whether I have write access to animint. But this PR is also checked in tdhock/animint, which also failed. You can see the checks.

**Comment by cpsievert** [comment]
@tdhock could you check to see if @caijun has write access and grant it?

This has my +1 since that error is to be expected.

**Comment by tdhock** [comment]
@caijun does have write access. Just add some info to the NEWS file and then feel free to merge with master.

**Comment by cpsievert** [comment]
Alrighty, @caijun in the future, please send pull requests from this remote (tdhock) to avoid that error

**Comment by caijun** [comment]
Instead of adding GITHUB_PAT env to .travis.yml, I set it in the settings of my Travis-CI; however, it still can not solve the problem of `Error: animint2gist() returns an object of class 'gist'`. At last, I gave up to try settting GITHUB_PAT env and merged the branch.

**Comment by cpsievert** [comment]
As I mentioned above, you can't do anything to resolve the error because this PR is _sent from your fork_. From [the Travis CI docs](http://docs.travis-ci.com/user/environment-variables/):

> Please note that secure env variables are not available for pull requests from forks. This is done due to the security risk of exposing such information in submitted code. Everyone can submit a pull request and if an unencrypted variable is available there, it could be easily displayed.

In the future, if you start a pull requests using the tdhock's remote, we'll avoid this issue.

**Comment by caijun** [comment]
Sorry, I hadn't understood the meaning of starting a pull requests using tdhock's remote before you made it clear again. Thanks for your information that helps to avoid the build error.

**Pull request merged**